### DISCUSSION TITLE
Allow configuring DAG documentation to be shown by default

### DISCUSSION THREAD

**Discussion by Limess** [apache/airflow#24167] [Ideas]
**Description**

When using `doc_md` to document DAGs the documentation is hidden behind an expanded by default in airflow 2.1.2.

This was not the case in 1.x.x where it was always shown, the documentation is now much less immediately visible.

**Use case / motivation**

Surfacing documentation to users more immediately. We write fairly lengthy markdown documentation for DAGs and since upgrading it's a few clicks away from the user when visiting a DAG which is new to the engineer or rarely trodden.

I'd be fine if this was a user level opt-in, DAG, DAG level, or top-level.

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
Maybe that should be a sticky setting in cookies ? @bbovenzi - @pierrejeambrun ? That **might** be simple to implement in pure React/JS

**Comment by pierrejeambrun** [comment]
It looks like this collapsible is for now implemented at the template level, and imported so it can appear across different pages of the UI. (Grid, Graph, Calendar...). Just saying that the implementation might not be grid/react specific.

Theoretically persisting this setting in a cookie or the local storage is perfectly doable. We could for instance default to showing the doc, and if a user hides it once, this would persist. (seems like a nice improvement)

Unfortunately I am not really familiar with our cookie management workflow, interacting with FAB. It is hard for me to estimate the effort that it would require to implement.

I am sure @bbovenzi will have more info about that :)

**Reply by bbovenzi** [reply]
The logic lives [here](https://github.com/apache/airflow/blob/main/airflow/www/templates/appbuilder/dag_docs.html). Someone could probably add some JS to the macro to read/add/remove a key from localStorage
